code/obj/item/mops_cleaners.dm

```diff
@@ -1192,3 +1194,72 @@ TYPEINFO(/obj/item/handheld_vacuum/overcharged)
 		animate(alpha=0, time=0.21 SECONDS, easing=SINE_EASING)
 		SPAWN(0.5 SECONDS)
 			qdel(src)
+
+
+/obj/item/trash_bag
+	name = "trash bag"
+	desc = "A flimsy bag for filling with things that are no longer wanted."
+	icon = 'icons/obj/janitor.dmi'
+	inhand_image_icon = 'icons/mob/inhand/hand_tools.dmi'
+	icon_state = "trashbag-f"
+	item_state = "trashbag-f"
+	uses_multiple_icon_states = TRUE
+	w_class = W_CLASS_TINY
+	rand_pos = TRUE
+	flags = FPRINT | TABLEPASS | NOSPLASH
+	tooltip_flags = REBUILD_DIST
+	var/base_state = "trashbag"
+	var/clothing_type = /obj/item/clothing/under/gimmick/trashsinglet
+
+	New()
+		..()
+		src.create_storage(/datum/storage/no_hud, prevent_holding = list(/obj/item/trash_bag), max_wclass = W_CLASS_NORMAL, slots = 20,
+			params = list("use_inventory_counter" = TRUE, "variable_weight" = TRUE, "max_weight" = 20))
+
+	attackby(obj/item/I, mob/user)
+		if (issnippingtool(I))
+			var/action = tgui_input_list(usr, "What do you want to do with [src]?", "Trash Bag", list("Cut into an outfit", "Add to contents"))
+			if (action == "Cut into an outfit")
+				boutput(user, "You begin cutting up [src].")
+				if (!do_after(user, 3 SECONDS))
+					boutput(user, "<span class='alert'>You were interrupted!</span>")
+					return
+				else
+					var/obj/item/clothing/under/gimmick/trashsinglet/trash_outfit = new clothing_type(get_turf(src))
+					playsound(src.loc, 'sound/items/Scissor.ogg', 100, 1)
+					user.u_equip(src)
+					if (length(src.storage.get_contents()))
+						for(var/obj/item/contents as anything in src.storage.get_contents())
+							src.storage.transfer_stored_item(contents, trash_outfit, TRUE)
+					usr.put_in_hand_or_drop(trash_outfit)
+					qdel(src)
+					return
+		..()
+
+	update_icon(mob/user)
+		if (!src.storage || !length(src.storage.get_contents()))
+			src.icon_state = initial(src.icon_state)
+			src.item_state = ""
```

Comment:
"" or initial(src.item_state)? Currently initialized as something other than ""

Reply:
TBH I have no idea what this is for and why you'd want the item state to be blank, it was in the original code.